src/mito2/src/compaction/compactor.rs

```diff
@@ -311,7 +311,7 @@ impl Compactor for DefaultCompactor {
                         file_size: sst_info.file_size,
                         available_indexes: {
                             let mut indexes = SmallVec::new();
-                            if sst_info.index_metadata.inverted_index.available {
+                            if sst_info.index_metadata.inverted_index.index_size > 0 {
```

Comment:
non-blocking: We can add a `is_available()` method that checks the index size.
```suggestion
                            if sst_info.index_metadata.inverted_index.is_available() {
```